src/cata_tiles.cpp

```diff
@@ -859,26 +860,30 @@ bool cata_tiles::draw_tile_at(tile_type *tile, int x, int y, int rota)
     return true;
 }
 
-bool cata_tiles::draw_lighting(int x, int y, LIGHTING l)
+bool cata_tiles::draw_lighting(int x, int y, lit_level l)
 {
     std::string light_name;
     switch(l) {
-        case HIDDEN:
+        case LL_BLANK:
             light_name = "lighting_hidden";
             break;
-        case LIGHT_NORMAL:
-            light_name = "lighting_lowlight_light";
-            break;
-        case LIGHT_DARK:
-            light_name = "lighting_lowlight_dark";
-            break;
-        case BOOMER_NORMAL:
-            light_name = "lighting_boomered_light";
+        case LL_BRIGHT_ONLY:
+        case LL_LIT:
```

Comment:
The code to check the other values came forward from the previous implementation. I'm not 100% sure that my restriction to only calling this in certain cases is correct. This could get shortened in a future revision, but until I know my behavior matches the original (or the correct, where they differ), I'll leave this in.

As to the return value, you're right. I have no idea why it was written to always return false. I imagine it used to be smarter, back before a lot of lighting-related falloff code seems to have been removed and commented out.

I'll see about doing a little bit of refactor here after I've got more testing done.